src/metabase/driver/sql/query_processor.clj

```diff
@@ -1372,10 +1372,18 @@
 (defn- apply-source-query
   "Handle a `:source-query` clause by adding a recursive `SELECT` or native query. At the time of this writing, all
   source queries are aliased as `source`."
-  [driver honeysql-form {{:keys [native params], :as source-query} :source-query}]
+  [driver honeysql-form {{:keys [native params],
+                          persisted :persisted-info/native
+                          :as source-query} :source-query}]
   (assoc honeysql-form
-         :from [[(if native
+         :from [[(cond
+                   native
                    (sql-source-query native params)
+
+                   persisted
+                   (sql-source-query persisted nil)
```

Comment:
This is the fix. We substitute the native sql in as we are emitting sql rather than earlier where it messes up aliases and field info.

Straight swap of `select <fields> from public.products` -> `select <fields> from cache_schema.cached_model`.

Reply:
Can `persisted` and  `native` both be true? I think `persisted` should be above it to be safe

Reply:
Yes i suppose it can. Thank you